core/Playlists/Playlist.vala

```diff
@@ -25,18 +25,18 @@
  *
  * Authored by: Corentin Noël <corentin@elementary.io>
  */
- 
+
 public abstract class Noise.Playlist : Object {
     public signal void request_play ();
-    public virtual Gee.ArrayQueue<Media> medias { get; internal set; }
+    public virtual Gee.ArrayList<Media> medias { get; internal set; }
```

Comment:
I think here you can change it like: 
```
get;
internal set;
default = new Gee.ArrayList<Media> ();
``` 

This way we avoid having the code in the constructor. You could use a similar pattern to all the code in the constructor.

Reply:
Also `media` is the plural word for `medium`. So there is no such thing as `medias`, perhaps the variable should be set to `media` and the class name to `Medium`?

Reply:
It's a big change, I would prefer to do it in a dedicated PR.